### PR TITLE
Update global-workflow and subcomponents to Hera/Rocky 8 partition

### LINKED ISSUE #2329: Migrate to Rocky8 spack-stack installations on Hera
### What new functionality do you need?

Hera is upgrading to the Rocky8 Linux OS, which requires a new spack-stack installation on the platform.  The global workflow, subcomponents, and external dependencies will need to be recompiled and tested on the platform before full transition in mid-April.

### What are the requirements for the new functionality?

- [x] Spack-stack v1.6.0 installation
- All submodules transitioned to Rocky8
   - [ ] UFS
   - [x] GSI
   - [x] GDAS
   - [ ] GSI-Utils
   - [ ] GSI-Monitor
   - [ ] EMC_verif-global
   - [ ] UFS_Utils
   - [ ] GFS-Utils

### Acceptance Criteria

The global workflow is able to reproduce results of RHEL.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
# Description

This PR addresses issue #2329. The following is accomplished:

- All submodule RDHPCS Hera stacks are updated to be compatible with the Rocky-8 distro spack-stack;
- The global-workflow version files `versions/build.hera.ver` and `versions/run.hera.ver` are updated for Rocky-8;
- All submodule hashes have been updated to be compliant with the Rocky-8 distro spack-stack (see the reference PRs below);
- Update to `parm/config/config.base` is made for not yet compliant packages;
- Relevant updates are made to `modulefiles/module_base.hera.lua` and `modulefiles/module_gwsetup.lua`.

  Resolves #2329
  Refs: [#958](https://github.com/NOAA-EMC/GDASApp/issues/958)
           [#49](https://github.com/NOAA-EMC/gfs-utils/issues/49)
           [#124](https://github.com/NOAA-EMC/GSI-Monitor/issues/124)
           [#31](https://github.com/NOAA-EMC/GSI-utils/issues/31)
           [#2167](https://github.com/ufs-community/ufs-weather-model/issues/2167)
           [#2143](https://github.com/ufs-community/ufs-weather-model/issues/2143)
           [#913](https://github.com/ufs-community/UFS_UTILS/issues/913)

# Type of change

- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

Manually tested for CI/CD PR application tests. A formal CI/CD will be performed upon PR.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+[submodule "sorc/gsi_utils.fd"]
+	path = sorc/gsi_utils.fd
+	url = https://github.com/HenryWinterbottom-NOAA/GSI-utils
+	branch = feature/gwdev_issue_2329
 [submodule "sorc/gsi_enkf.fd"]
-  path = sorc/gsi_enkf.fd
-  url = https://github.com/NOAA-EMC/GSI.git
+	path = sorc/gsi_enkf.fd
+	url = https://github.com/HenryWinterbottom-NOAA/GSI
+	branch = feature/gwdev_issue_2329
```
Comment (aerorahul) [inline .gitmodules]: GSI has been updated to Rocky 8
```suggestion
  path = sorc/gsi_enkf.fd
  url = https://github.com/NOAA-EMC/GSI.git

```

File: parm/config/gfs/config.base [hunk 2]
```diff
@@ -383,8 +383,8 @@ export FITSARC="YES"
 export FHMAX_FITS=132
 [[ "${FHMAX_FITS}" -gt "${FHMAX_GFS}" ]] && export FHMAX_FITS=${FHMAX_GFS}
 
-# The monitor jobs are not yet supported for JEDIATMVAR
-if [[ ${DO_JEDIATMVAR} = "YES" ]]; then
+# The monitor jobs are not yet supported for JEDIATMVAR or RDHPCS Hera for Rocky-8.
+if [[ ${DO_JEDIATMVAR} = "YES" ]] || [[ ${machine} = "HERA" ]]; then
    export DO_VERFOZN="NO"     # Ozone data assimilation monitoring
    export DO_VERFRAD="NO"     # Radiance data assimilation monitoring
    export DO_VMINMON="NO"     # GSI minimization monitoring
```
Comment (DavidHuber-NOAA) [inline parm/config/gfs/config.base]: The ozone and radiance monitor are supported on Hera, so only the minimization monitor needs to be disabled for now.

```suggestion
# The monitor jobs are not yet supported for JEDIATMVAR.
if [[ ${DO_JEDIATMVAR} = "YES" ]]; then
   export DO_VERFOZN="NO"     # Ozone data assimilation monitoring
   export DO_VERFRAD="NO"     # Radiance data assimilation monitoring
   export DO_VMINMON="NO"     # GSI minimization monitoring
# The minimization monitor is not yet supported on RDHPCS Hera for Rocky-8
elif [[ ${machine} = "HERA" ]]; then
   export DO_VMINMON="NO"     # GSI minimization monitoring

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
 [submodule "sorc/gsi_enkf.fd"]
   path = sorc/gsi_enkf.fd
   url = https://github.com/NOAA-EMC/GSI.git
 [submodule "sorc/gdas.cd"]
   path = sorc/gdas.cd
   url = https://github.com/NOAA-EMC/GDASApp.git
-[submodule "sorc/gsi_utils.fd"]
-  path = sorc/gsi_utils.fd
-  url = https://github.com/NOAA-EMC/GSI-Utils.git
+[submodule "sorc/verif-global.fd"]
```
Comment (aerorahul) [inline .gitmodules]: can the order and the indentation be reverted to `develop` so only the new submodule added (`upp`) is seen in the diff?

File: modulefiles/module_base.hera.lua [hunk 4]
```diff
@@ -41,11 +41,10 @@ load(pathJoin("py-xarray", (os.getenv("py_xarray_ver") or "None")))
 setenv("WGRIB2","wgrib2")
 setenv("UTILROOT",(os.getenv("prod_util_ROOT") or "None"))
 
---prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/global/glopara/git/prepobs/v" .. (os.getenv("prepobs_run_ver") or "None"), "modulefiles"))
```
Comment (KateFriedman-NOAA) [inline modulefiles/module_base.hera.lua:44]: Can you leave the commented out prepobs module line for now? It's there for if/when we go back to versioned installs. We can remove this line later if we move to make prepobs a submodule instead. Leave this line here for now, thanks!

File: modulefiles/module_base.hera.lua [hunk 5]
```diff
@@ -41,11 +41,10 @@ load(pathJoin("py-xarray", (os.getenv("py_xarray_ver") or "None")))
...
-prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/global/glopara/git/prepobs/feature-GFSv17_com_reorg_log_update/modulefiles"))
+prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/global/glopara/git/prepobs/feature-hera-rocky8/modulefiles"))
```
Comment (KateFriedman-NOAA) [inline modulefiles/module_base.hera.lua]: We have completed the prepobs PR into our prepobs `dev/gfsv17` branch. Please update this to change `feature-hera-rocky8` to `dev-gfsv17`.

For reference, the full install path is: `/scratch1/NCEPDEV/global/glopara/git/prepobs/dev-gfsv17`

File: sorc/link_workflow.sh [hunk 6]
```diff
@@ -345,10 +345,11 @@ fi
 #--link source code directories
 #------------------------------
 cd "${HOMEgfs}/sorc" || exit 8
-if [[ -d ufs_model.fd ]]; then
-  [[ -d upp.fd ]] && rm -rf upp.fd
-  ${LINK} ufs_model.fd/FV3/upp upp.fd
-fi
+# TODO: Commenting out until UPP is up-to-date with Rocky-8.
```
Comment (KateFriedman-NOAA) [inline sorc/link_workflow.sh:355]: In this comment, do you mean "..until the UPP _**in UWM**_ is up-to-date..."? Wanna avoid confusion with what we're waiting on for later.

File: .gitmodules (hunk 3 above)
Comment (KateFriedman-NOAA) [reply .gitmodules]: I second this request.

File: modulefiles/module_base.hera.lua [hunk 7]
```diff
@@ -41,11 +41,10 @@ load(pathJoin("py-xarray", (os.getenv("py_xarray_ver") or "None")))
...
 load(pathJoin("prepobs", (os.getenv("prepobs_run_ver") or "None")))
 
-prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/global/glopara/git/Fit2Obs/v" .. (os.getenv("fit2obs_ver") or "None"), "modulefiles"))
+prepend_path("MODULEPATH", pathJoin("/scratch1/NCEPDEV/global/glopara/git/Fit2Obs/feature-hera-rocky8/modulefiles"))
```
Comment (KateFriedman-NOAA) [inline modulefiles/module_base.hera.lua]: Alrighty, so we merged the Fit2Obs PR with the Rocky8 changes for Hera, I cut a version tag, and installed it here on Hera: `/scratch1/NCEPDEV/global/glopara/git/Fit2Obs/v1.1.0`.

Please revert this back to the prior line that includes `fit2obs_ver` and then update `fit2obs_ver` in `run.spack.ver` to be `export fit2obs_ver=1.1.0`. I will install this new tag on our platforms on Monday.
https://github.com/NOAA-EMC/global-workflow/blob/develop/versions/run.spack.ver#L34

**Review by KateFriedman-NOAA — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 8]
```diff
@@ -1,28 +1,30 @@
...
-  url = https://github.com/NOAA-EMC/gfs-utils
-[submodule "sorc/ufs_utils.fd"]
-  path = sorc/ufs_utils.fd
-  url = https://github.com/ufs-community/UFS_UTILS.git
-[submodule "sorc/verif-global.fd"]
-  path = sorc/verif-global.fd
-  url = https://github.com/NOAA-EMC/EMC_verif-global.git
 [submodule "sorc/gsi_enkf.fd"]
   path = sorc/gsi_enkf.fd
   url = https://github.com/NOAA-EMC/GSI.git
```
Comment (aerorahul) [inline .gitmodules]: The GSI hash in this PR does not point to a commit in the authoritative repo. 
Screenshot 2024-03-22 at 2 51 54 PM

File: .gitmodules (hunk 3 above)
Comment (HenryRWinterbottom) [reply .gitmodules]: Done.

File: modulefiles/module_base.hera.lua (hunk 4 above)
Comment (HenryRWinterbottom) [reply modulefiles/module_base.hera.lua:44]: Done.

File: .gitmodules (hunk 8 above)
Comment (HenryRWinterbottom) [reply .gitmodules]: Updated hash to [dfb958fa9](https://github.com/NOAA-EMC/GSI/commit/dfb958fa9372c10c808d20e64f2955652017ee32).

File: .gitmodules (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply .gitmodules]: This has not been done, or at least not pushed.

File: .gitmodules (hunk 8 above)
Comment (KateFriedman-NOAA) [reply .gitmodules]: @HenryWinterbottom-NOAA Please reset the order of the submodules in this file so it shows just the addition of the upp submodule. Thanks!

File: .gitmodules (hunk 3 above)
Comment (HenryRWinterbottom) [reply .gitmodules]: In my [branch](https://github.com/HenryWinterbottom-NOAA/global-workflow/blob/feature/gwdev_issue_2329/.gitmodules):

I see the following:

```
[submodule "sorc/wxflow"]
  path = sorc/wxflow
  url = https://github.com/NOAA-EMC/wxflow
[submodule "sorc/gfs_utils.fd"]
  path = sorc/gfs_utils.fd
  url = https://github.com/NOAA-EMC/gfs-utils
[submodule "sorc/gsi_monitor.fd"]
  path = sorc/gsi_monitor.fd
  url = https://github.com/NOAA-EMC/GSI-Monitor
[submodule "sorc/ufs_utils.fd"]
  path = sorc/ufs_utils.fd
  url = https://github.com/ufs-community/UFS_UTILS
[submodule "sorc/gsi_utils.fd"]
  path = sorc/gsi_utils.fd
  url = https://github.com/NOAA-EMC/GSI-utils
[submodule "sorc/gsi_enkf.fd"]
  path = sorc/gsi_enkf.fd
  url = https://github.com/NOAA-EMC/GSI.git
[submodule "sorc/verif-global.fd"]
  path = sorc/verif-global.fd
  url = https://github.com/NOAA-EMC/EMC_verif-global.git
[submodule "sorc/gdas.cd"]
  path = sorc/gdas.cd
  url = https://github.com/NOAA-EMC/GDASApp.git
[submodule "sorc/ufs_model.fd"]
  path = sorc/ufs_model.fd
  url = https://github.com/ufs-community/ufs-weather-model
[submodule "sorc/upp.fd"]
  path = sorc/upp.fd
  url = https://github.com/NOAA-EMC/UPP
```

Comment (WalterKolczynski-NOAA) [reply .gitmodules]: That is not the order they appear in develop: https://github.com/NOAA-EMC/global-workflow/blob/develop/.gitmodules

File: modulefiles/module_base.hera.lua (hunk 7 above)
Comment (KateFriedman-NOAA) [reply modulefiles/module_base.hera.lua]: Looks good, thanks!

**Comment by WalterKolczynski-NOAA** [comment]
GEFS test needs to be skipped when using bash CI.

**Comment by WalterKolczynski-NOAA** [comment]
I think gdas experiments need to wait for #2435

**Comment by WalterKolczynski-NOAA** [comment]
#2435 has been merged. Please update this branch and we'll try again.

**Comment by RussTreadon-NOAA** [comment]
As I test I installed `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` in `/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr2421`.  The working copy was update with the current head of g-w `develop`.

The following components failed to build
```
build_ufs.sh failed with status 1!
build_upp.sh failed with status 2!
```

JEDI ATM CI was configured.  Since the UFS model was not built, I primed the test with forecast files generated from a pre-Rocky8 test.  After doing this the JEDI ATM CI was manually booted from 202402400 gdas and gfs prep.  The JEDI ATM gfs and gdas variational jobs ran OK.  The JEDI ATM local ensemble jobs ran OK.

Note that I built `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` using the current heads of GDASApp `develop`, [2198b41](https://github.com/NOAA-EMC/GDASApp/commit/2198b419567cf7efa7404cd076e76e01d86f9e58), and GSI `develop`, [b53740a](https://github.com/NOAA-EMC/GSI/commit/b53740a7bd1cc416f634589075b8c8b89f0ef761).   

Using the current head of GSI `develop` is important.  A recent commit to GSI `develop` fixed nsst and threading bugs.

**Comment by KateFriedman-NOAA** [comment]
> Note that I built HenryWinterbottom-NOAA:feature/gwdev_issue_2329 using the current heads of GDASApp develop, [2198b41](https://github.com/NOAA-EMC/GDASApp/commit/2198b419567cf7efa7404cd076e76e01d86f9e58), and GSI develop, [b53740a](https://github.com/NOAA-EMC/GSI/commit/b53740a7bd1cc416f634589075b8c8b89f0ef761).
Using the current head of GSI develop is important. A recent commit to GSI develop fixed nsst and threading bugs.

@RussTreadon-NOAA thanks for testing and reporting these build issues, as well as letting us know we should move to the newer GDASApp and GSI hashes.

@HenryWinterbottom-NOAA please sync your branch with global-workflow `develop` and update the GDASApp and GSI hashes to their latest ones (as outlined by Russ). Then please retest cloning, build, link on Hera. If there are build issues, they will need to be resolved. Once resolved, the CI tests can be retried. Thanks!

**Comment by KateFriedman-NOAA** [comment]
To help out, I cloned a copy of Henry's branch for this PR, synced it with `develop`, and ran the build. I did not encounter any build failures. The sync merge updated the UWM hash, which may have corrected the build issues Russ saw (?). I doublechecked I was on a Rocky8 login node for the build. See output from sync and build below, as well as path to my test clone on Hera.

I did not update the GDASApp or GSI hashes for this build test. Will let @HenryWinterbottom-NOAA do so and report on build success/failure.

```
[Kate.Friedman@hfe10 sorc]$ pwd
/scratch1/NCEPDEV/global/Kate.Friedman/git/feature-gwdev_issue_2329/sorc
[Kate.Friedman@hfe10 feature-gwdev_issue_2329]$ git merge upstream/develop
Auto-merging sorc/link_workflow.sh
Note: Fast-forwarding submodule sorc/ufs_model.fd to 7fdb58cad0dad2f62ce7813c6719554d1c5a17af
hint: Waiting for your editor to close the file... 
Merge remote-tracking branch 'upstream/develop' into feature/gwdev_issue_2329

# By David Huber (1) and others
# Via GitHub
* upstream/develop:
  Add option to use traditional threading in the UFS (#2384)
  Correct GDASApp paths (#2435)
  Fix *earc jobs where the number of members isn't a multiple of 10 (#2424)
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.
~                                                                                                                                        
...                                                                                                                                      
~                                                                                                                                        
".git/MERGE_MSG" 13L, 539C written
Merge made by the 'ort' strategy.
 jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP |  3 ++-
 jobs/rocoto/prepatmiodaobs.sh         |  5 +++--
 jobs/rocoto/prepsnowobs.sh            |  4 +++-
 parm/config/gefs/config.fcst          |  2 ++
 parm/config/gefs/config.resources     | 18 ++++++++++++++++--
 parm/config/gefs/config.ufs           | 31 ++++++++++++++++++++++++-------
 parm/config/gfs/config.fcst           |  2 ++
 parm/config/gfs/config.resources      | 27 +++++++++++++++++++++++----
 parm/config/gfs/config.ufs            | 33 +++++++++++++++++++++++++--------
 parm/ufs/fv3/diag_table_aod           |  2 +-
 scripts/exglobal_forecast.sh          |  6 ++++++
 sorc/link_workflow.sh                 | 11 +++++++++--
 ush/hpssarch_gen.sh                   |  2 +-
 ush/module-setup.sh                   |  8 ++++----
 ush/preamble.sh                       | 16 ++++++++++++++++
 15 files changed, 137 insertions(+), 33 deletions(-)
[Kate.Friedman@hfe10 feature-gwdev_issue_2329]$ git status
On branch feature/gwdev_issue_2329
Your branch is ahead of 'origin/feature/gwdev_issue_2329' by 4 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
[Kate.Friedman@hfe10 feature-gwdev_issue_2329]$ cd sorc/
[Kate.Friedman@hfe10 sorc]$ ./build_all.sh -g
Creating logs folder
Creating /scratch1/NCEPDEV/global/Kate.Friedman/git/feature-gwdev_issue_2329/exec folder
Running "module reset". Resetting modules to system default. The following $MODULEPATH directories have been removed: /contrib/anaconda/modulefiles /scratch2/BMC/gsienkf/UFS-RNR/UFS-RNR-stack/modules
Building gsi_enkf, ufs, gfs_utils, ww3prepost, ufs_utils, gsi_utils, gsi_monitor, upp
Starting build_gsi_enkf.sh
Starting build_ufs.sh
Starting build_gfs_utils.sh
Starting build_ww3prepost.sh
Starting build_ufs_utils.sh
Starting build_gsi_utils.sh
Starting build_gsi_monitor.sh
Starting build_upp.sh
build_gfs_utils.sh completed successfully!
build_ufs_utils.sh completed successfully!
build_gsi_monitor.sh completed successfully!
build_ww3prepost.sh completed successfully!
build_gsi_utils.sh completed successfully!
build_upp.sh completed successfully!
build_gsi_enkf.sh completed successfully!
build_ufs.sh completed successfully!

 .... Build system finished .... 
[Kate.Friedman@hfe10 sorc]$ ./link_workflow.sh 
Begin link_workflow.sh at Wed Mar 27 14:34:58 UTC 2024
End link_workflow.sh at 14:34:58 with error code 0 (time elapsed: 00:00:00)
```

**Comment by HenryRWinterbottom** [comment]
> As I test I installed `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` in `/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/pr2421`. The working copy was update with the current head of g-w `develop`.
> 
> The following components failed to build
> 
> ```
> build_ufs.sh failed with status 1!
> build_upp.sh failed with status 2!
> ```
> 
> JEDI ATM CI was configured. Since the UFS model was not built, I primed the test with forecast files generated from a pre-Rocky8 test. After doing this the JEDI ATM CI was manually booted from 202402400 gdas and gfs prep. The JEDI ATM gfs and gdas variational jobs ran OK. The JEDI ATM local ensemble jobs ran OK.
> 
> Note that I built `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` using the current heads of GDASApp `develop`, [2198b41](https://github.com/NOAA-EMC/GDASApp/commit/2198b419567cf7efa7404cd076e76e01d86f9e58), and GSI `develop`, [b53740a](https://github.com/NOAA-EMC/GSI/commit/b53740a7bd1cc416f634589075b8c8b89f0ef761).
> 
> Using the current head of GSI `develop` is important. A recent commit to GSI `develop` fixed nsst and threading bugs.

@RussTreadon-NOAA I updated the hashes for `gsi_enkf.fd` and `gdas.cd`. I did not have any issues building any of the g-w submodules. 

Can you try again?

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres @RussTreadon-NOAA This is using GDASapp develop from a couple days ago and still has the issue with the missing obs types causing the whole thing to fail. We need this to be able to gracefully degrade when certain obs types aren't available.

**Comment by WalterKolczynski-NOAA** [comment]
Going to turn off the 3DVarAOWCDA test until the problem is resolved.

**Comment by RussTreadon-NOAA** [comment]
Fresh clone of `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` shows 
- `gsi_enkf.fd` is GSI hash b53740a7b
- `gdas.cd` is GDASApp hash 2198b41

These are the hashes which were tested this morning.  They built without problem this morning.  They should build OK now.

The failed builds  from this morning were for upp and the ufs weather model.

**Comment by RussTreadon-NOAA** [comment]
> @guillaumevernieres @RussTreadon-NOAA This is using GDASapp develop from a couple days ago and still has the issue with the missing obs types causing the whole thing to fail. We need this to be able to gracefully degrade when certain obs types aren't available.

Tag @AndrewEichmann-NOAA since he works on JEDI marine DA.

**Comment by WalterKolczynski-NOAA** [comment]
> Fresh clone of `HenryWinterbottom-NOAA:feature/gwdev_issue_2329` shows
> 
> * `gsi_enkf.fd` is GSI hash b53740a7b
> * `gdas.cd` is GDASApp hash 2198b41
> 
> These are the hashes which were tested this morning. They built without problem this morning. They should build OK now.
> 
> The failed builds from this morning were for upp and the ufs weather model.

The build is completing, but the test is failing because obs files aren't being created because some of the new obs types aren't in dump. (Based on my understanding from comments in other PRs.)
```
2024-03-27 18:07:24,736 - INFO     - file_utils  : Copied adt_rads_all2ioda.yaml to /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2421/RUNTESTS/COMR
OOT/C48mx500_3DVarAOWCDA_e9ff820f/gdas.20210324/18/obs/adt_rads_all2ioda.yaml
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/fsutils.py", line 83, in cp
    shutil.copy2(source, target)
  File "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/python-3.10.13-2p2rady/lib/python3.10/shu
til.py", line 434, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/intel/2021.5.0/python-3.10.13-2p2rady/lib/python3.10/shu
til.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'gdas.t18z.sst_avhrr_ma_l3u.2021032418.nc4'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2421/global-workflow/ush/exglobal_prep_ocean_obs.py", line 129, in <module>
    FileHandler({'copy': files_to_save}).sync()
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/file_utils.py", line 43, in sync
    sync_factory[action](files)
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/file_utils.py", line 63, in _copy_files
    cp(src, dest)
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/fsutils.py", line 85, in cp
    raise OSError(f"unable to copy {source} to {target}")
OSError: unable to copy gdas.t18z.sst_avhrr_ma_l3u.2021032418.nc4 to /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/2421/RUNTESTS/COMROOT/C48mx500_3D
VarAOWCDA_e9ff820f/gdas.20210324/18/obs/gdas.t18z.sst_avhrr_ma_l3u.2021032418.nc4
```

**Comment by RussTreadon-NOAA** [comment]
@WalterKolczynski-NOAA , I primarily work on JEDI ATM DA, not JEDI Marine DA.  The successful tests I reported this morning were me manually toggling on JEDI ATM CI case `C96C48_ufs_hybatmDA`.  JEDI ATM bufr2ioda conversion and var/ens init, run, and final work.  I did not try running JEDI Marine.

You are right.  Not all bufr2ioda converters cleanly handle the variety of things which can happen in operations:  missing dump files, zero length dump files, missing subsets, etc.  I added some logic to JEDI ATM bufr2ioda converters to address these operational realities.  A more thorough and, hopefully, more consistent approach was left for the DA observation team to discuss and implement.

**Comment by AndrewEichmann-NOAA** [comment]
@WalterKolczynski-NOAA @RussTreadon-NOAA It might be that the the IODA converter for that obs type is failing - the log for that test seems to be deleted, is there an existing one?

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA @RussTreadon-NOAA It might be that the the IODA converter for that obs type is failing - the log for that test seems to be deleted, is there an existing one?

The one from my other PR is still around:
```
/scratch1/NCEPDEV/global/CI/2413/RUNTESTS/COMROOT/C48mx500_3DVarAOWCDA_6aedab21/logs/2021032418/gdasprepoceanobs.log
```

Terry is working on ways for the log not to get clobbered when we restart tests.

[gdasprepoceanobs.log](https://gist.github.com/emcbot/a7f97e26378e2842289d394de3ec64e2)

**Comment by guillaumevernieres** [comment]
> @WalterKolczynski-NOAA @RussTreadon-NOAA It might be that the the IODA converter for that obs type is failing - the log for that test seems to be deleted, is there an existing one?

It's not "might", it's exactly that and something we already talked about a few days ago. I can submit a temporary fix tomorrow, but that's the bug that Mindo is working on @AndrewEichmann-NOAA . All that needs to be done is to comment out the obs that are failing ...

**Comment by WalterKolczynski-NOAA** [comment]
Related follow-up issues:
- #2437 
- #2438 
- #2439

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**